snap/snapcraft.yaml

```diff
@@ -113,65 +109,62 @@ parts:
     prime:
       - usr/lib
       - usr/share/X11
-      - -usr/share/doc
-      - -usr/share/lintian
-      - -usr/share/man
 
   wayland:
     # Wayland support (not much cost to having this)
+    after: [i386-repo]  # because LP#2009278
     plugin: nil
     stage-packages:
       - libwayland-client0
       - libwayland-egl1
       - libwayland-server0
+      - libnvidia-egl-wayland1
     prime:
       - usr/lib
-      - -usr/share/doc
+      - usr/share/egl/egl_external_platform.d
 
   apis-i386:
       # This provides the essential APIs
+      #   o libGL.so.1
       #   o libEGL.so.1
       #   o libva.so.2
       #   o libvulkan.so.1
-      #   o libGLX.so.0
+      #   o libgbm.so.1
       #
-      plugin: nil
-      stage-packages:
-        - on amd64:
-          - libegl1:i386
-          - libgles2:i386
-          - libvulkan1:i386
-          - libglx0:i386
-      override-prime: |
-        if [ `arch` = "x86_64" ]; then craftctl default; fi
-      prime:
-        - usr/lib
-        - usr/share/glvnd
-        - -usr/share/X11
-        - -usr/share/bug
-        - -usr/share/doc
-        - -usr/share/lintian
-        - -usr/share/man
-        - etc/glvnd
-        - -etc/vulkan
+    after: [i386-repo]  # because LP#2009278
+    plugin: nil
+    stage-packages:
+      - on amd64:
+        - libgl1:i386
+        - libegl1:i386
+        - libgles2:i386
+        - libvulkan1:i386
+        - libgbm1:i386
```

Comment:
do you want to add `:armhf` packages on arm64 builds?

Reply:
We had discussed this, but didn't have an immediate use case for it - and since this is something we can do entirely on the provider side, decided to punt it to when we do find a use (the use for x86 is primarily games / legacy software).

We could certainly mirror the behaviour on ARM, if we found that appropriate.